src/Http/Http.Extensions/gen/StaticRouteHandlerModel/StaticRouteHandlerModel.cs

```diff
@@ -0,0 +1,20 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+namespace Microsoft.AspNetCore.Http.Generators.StaticRouteHandlerModel;
+
+internal enum RequestParameterSource
+{
+    Query,
+    Route,
+    Header,
+    Form,
+    Service,
+    BodyOrService,
+}
+
+internal record RequestParameter(string Name, string Type, RequestParameterSource Source, bool IsOptional, object? DefaultValue);
+internal record EndpointRoute(string RoutePattern, List<string> RouteParameters);
```

Comment:
`List<string>` is problematic here as it will compare unequal.

Reply:
CommunityToolkit uses https://github.com/CommunityToolkit/dotnet/blob/main/src/CommunityToolkit.Mvvm.SourceGenerators/Helpers/EquatableArray%7BT%7D.cs in generators. Would that work here?

Reply:
Consider removing it for now.

When it is needed in the future, care should be taken about equality.